modules/tensor_mechanics/src/vectorpostprocessors/InteractionIntegral.C

```diff
@@ -93,11 +103,24 @@ InteractionIntegralTempl<is_ad>::InteractionIntegralTempl(const InputParameters
     _fe_type(_fe_vars[0]->feType()),
     _grad_disp(3),
     _has_temp(isCoupled("temperature")),
+    _has_functionally_graded_youngs_modulus_crack_dir_gradient(
```

Comment:
This way works, but I think it's cleaner to not have these `_has_functionally_graded_youngs_modulus_crack_dir_gradient` and `_has_functionally_graded_youngs_modulus` bool member variables. You can just do the `isParamSetByUser()` check right in the initialization of those variables, and set them to `nullptr` if they're not defined, and then do your error checking on line 146 based on whether they're set to `nullptr`. If you want to leave it as-is, that's fine too.